pkg/protocol/tipmanager/v1/tip_metadata.go

```diff
@@ -0,0 +1,193 @@
+package tipmanagerv1
+
+import (
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/iota-core/pkg/core/promise"
+	"github.com/iotaledger/iota-core/pkg/model"
+	"github.com/iotaledger/iota-core/pkg/protocol/engine/blocks"
+	"github.com/iotaledger/iota-core/pkg/protocol/tipmanager"
+)
+
+// TipMetadata represents the metadata for a block in the TipManager.
+type TipMetadata struct {
+	// Block holds the actual block.
+	block *blocks.Block
+
+	// tipPool holds the TipPool the block is currently in.
+	tipPool *promise.Value[tipmanager.TipPool]
+
+	// stronglyConnectedChildren holds the number of strong children that can be reached from the tips using only strong
+	// references.
+	stronglyConnectedChildren *promise.Value[int]
+
+	// weaklyConnectedChildren holds the number of weak children that can be reached from the tips.
+	weaklyConnectedChildren *promise.Value[int]
+
+	// stronglyReferencedByTips is a derived property that is true if the block has at least one strongly connected
+	// child.
+	stronglyReferencedByTips *promise.Value[bool]
+
+	// referencedByTips is a derived property that is true if the block has at least one strongly or weakly connected
+	// child.
+	referencedByTips *promise.Value[bool]
+
+	// stronglyConnectedToTips is a derived property that is true if the block is either strongly referenced by tips or
+	// part of the strong TipPool.
+	stronglyConnectedToTips *promise.Value[bool]
```

Comment:
Yes, a block can be weakly and strongly connected to the tips at the same time, they are somewhat orthogonal properties and it can become false if no other block that is part of the corresponding TipPools references a block (mostly in respect to dropped blocks).

Reply:
Ok, so for this property to become `false` again, assuming that it has one child that belong to the tip set, we'd try to select the child, fail, drop it from the strong tip pool, then the block itself is added back to the tip set (but it's still `stronglyConnectedToTips` and once we fail to select it as a tip and drop it from the strong tip pool this property changes. Is the undestanding correct?

Reply:
Yes, that sounds correct.